tools/state-parts-dump/src/rpc_requests.rs

```diff
@@ -0,0 +1,214 @@
+use serde::Deserialize;
+use std::error::Error;
+
+#[derive(Deserialize)]
+struct ValidatorsResponse {
+    result: ValidatorsInfo,
+}
+
+#[derive(Deserialize)]
+struct BlockResponse {
+    result: BlockInfo,
+}
+
+#[derive(Deserialize)]
+struct StatusResponse {
+    result: StatusInfo,
+}
+
+#[derive(Deserialize)]
+struct StatusInfo {
+    sync_info: SyncInfo
+}
+
+#[derive(Deserialize)]
+struct SyncInfo {
+    latest_block_hash: String
+}
+
+#[derive(Deserialize)]
+struct ValidatorsInfo {
+    epoch_start_height: u64,
+    epoch_height: u64
+}
+
+#[derive(Deserialize)]
+struct BlockInfo {
+    header: BlockInfoHeader,
+    chunks: Vec<ChunkInfo>
+}
+
+#[derive(Deserialize)]
+struct BlockInfoHeader {
+    hash: String,
+    prev_hash: String,
+    next_epoch_id: String,
+    epoch_id: String
+}
+
+#[derive(Deserialize)]
+struct ChunkInfo {
+    prev_state_root: String,
+}
+
+pub struct RpcClient {
+    client: reqwest::blocking::Client,
+    rpc_endpoint: String,
+}
+
+impl RpcClient {
+    // network can only be "mainnet" or "testnet"
```

Comment:
Should be fine to accept the rpc_endpoint as an CLI parameter.
This doesn't seem bad to me:
`state-parts-dump --rpc-endpoint https://rpc.mainnet.near.org`